frontend/src/App.js

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import "./App.css";
+import mainMapImage from "./assets/images/usmap-background.jpg";
+import { map } from "./fakeData";
+import City from "./components/City";
+import ConnectionPipe from "./components/ConnectionPipe";
+import * as tools from "./functions";
+
+function App() {
+  // Prep map data
+  var citiesArray = [];
+  var connectionsArray = [];
+
+  for (var key in map.cities) {
+    // add all cities to an array
+    citiesArray.push(map.cities[key]);
+
+    // add all connections to an array
+    for (var connection in map.cities[key].connections) {
+      connectionsArray.push({
+        startCity: key,
+        endCity: map.cities[key].connections[connection].id,
+        cost: map.cities[key].connections[connection].cost
+      });
+    }
+  }
+
+  var cities = citiesArray.map((city, index) => {
+    return (
+      <City
+        key={index}
+        name={city.name}
+        id={city.id}
+        top={city.top}
+        left={city.left}
+      />
+    );
+  });
+
+  // Pipe Connections
+  // remove repeats
+  for (var i in connectionsArray) {
```

Comment:
nit: if there are 40 cities, then this is 1600 comparisons. May be fine in this case. But good to avoid nested loops like this as a general rule. Can avoid nested loops like this by creating a `usedConnectionSet` and then checking if new connections already exist in the set.

Reply:
Fixed this with: 

```    var uniqueConnectionsArray = [];

    for (var i in connectionsArray) {
      // create an array of duplicate connections
      var duplicatesArray = uniqueConnectionsArray.filter(
        // eslint-disable-next-line no-loop-func
        connection =>
          (connection.startCity === connectionsArray[i].startCity &&
            connection.endCity === connectionsArray[i].endCity) ||
          (connection.startCity === connectionsArray[i].endCity &&
            connection.endCity === connectionsArray[i].startCity)
      );
      // if no duplicates, then add new unique connection
      if (duplicatesArray.length === 0) {
        uniqueConnectionsArray.push(connectionsArray[i]);
      }
    }```